.eslintrc.js

```diff
@@ -87,9 +80,27 @@ module.exports = {
         },
       },
     ],
-    ...legacyCode,
+    // note you must disable the base rule as it can report incorrect errors
+    'no-use-before-define': 'off',
+    'react/jsx-filename-extension': ['error', { extensions: ['.js', '.tsx'] }],
+    'no-underscore-dangle': [
+      'error',
+      { allow: ['__REDUX_DEVTOOLS_EXTENSION__'] },
+    ],
   },
   overrides: [
+    // Legacy Code - add to `excludedFiles` when adopting desired rules in new code progressively
+    {
+      files: ['*.js', '*.ts', '*.jsx', '*.tsx'],
+      // add new (non-legacy) code to the exclusion here
+      excludedFiles: [
+        'client/src/api/**',
+        'client/src/config/**',
+        'client/src/includes/**',
+        'client/src/plugins/**',
+      ],
+      rules: legacyCode,
```

Comment:
I’m finding this hard to reason about – wouldn’t it be better for us to use a `files` pattern to target legacy code, rather than have to exclude new code?

Reply:
No problems - can rework to that direction

Reply:
I’m happy to go for what you have here if you prefer, just need to understand the trade-offs a bit more.